tls/s2n_server_key_exchange.c

```diff
@@ -45,13 +45,13 @@ int s2n_server_key_recv(struct s2n_connection *conn)
     GUARD(s2n_kex_server_key_recv_read_data(key_exchange, conn, &data_to_verify, &kex_data));
 
     /* Add common signature data */
-    if (conn->actual_protocol_version == S2N_TLS12) {
-        s2n_hash_algorithm hash_algorithm;
-        s2n_signature_algorithm signature_algorithm;
-        GUARD(s2n_get_signature_hash_pair_if_supported(in, &hash_algorithm, &signature_algorithm));
-        GUARD(s2n_hash_init(signature_hash, hash_algorithm));
+    if (conn->actual_protocol_version >= S2N_TLS12) {
```

Comment:
hmm, i'm not sure if there's any value changing this condition. this function should not be called in tls1.3 and we should consider guard the entries to be < TLS13 instead.

Reply:
Fixed